evm/vm/evm.py

```diff
@@ -125,10 +106,10 @@ def get_vm_class_for_block_number(cls, block_number):
         """
         Return the vm class for the given block number.
         """
-        range = find_range(tuple(cls.vms_by_range.keys()), block_number)
-        base_vm_class = cls.vms_by_range[range]
-        vm_class = base_vm_class.configure(db=cls.db)
```

Comment:
The only annoyance I encountered is that before this change, the `VM.get_block_class()` method would return a class with a `db` property set, and that was handy because we could just pass that to `rlp.decode()` to get a valid block, but now with this change we need to [pass the db as an extra keyword argument to rlp.decode()](https://github.com/gsalgado/py-evm/blob/issue-11/evm/vm/flavors/frontier/blocks.py#L144-L147)